### DISCUSSION TITLE
Is there any way to get the "filter name" inside the common custom filter function defined against it

### DISCUSSION THREAD

**Discussion by cs-satish-mishra** [pallets/jinja#1840] [Q&A] [answered]
I have written generic custom filters. 

```py
def common_custom_filter_func(**args, **kwargs):
      print(filter name)
```

And there are many filters with different names that are mapped to the same function 

```
{ 
"filter1": common_custom_filter_func, 
"filter2": common_custom_filter_func, 
}
```

Is there any way I can get my filter name inside the filter function when it's called? For example, I need to catch the name filter1 inside `common_custom_filter_func` when the below template is called. And respectively for filter2 

```
 {{ data | filter1 }}  ----> prints filter1 
 {{ data | filter2 }}  ----> prints filter2 
``` 

 I tried pass_context but it does not have any attribute that has the caller filter name.

**Comment by ThiefMaster** [comment, marked as answer]
No, that name isn't even known to Python itself since it's not the function name but just some dict key it's assigned to.

Here's a workaround:

```py
filters = { 
    "filter1": common_custom_filter_func, 
    "filter2": common_custom_filter_func, 
}
filters = {k: functools.partial(v, k) for k, v in filters.items()}
```

Then just add a new positional argument to be beginning of your function, e.g. `def common_filter(name, whatever, else)` and use that argument to get the name.

**Comment by cs-satish-mishra** [comment]
@ThiefMaster Thank you so much for the quick reply,   It works.
